/* Copyright 2022 The MediaPipe Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto2";

package mediapipe;

import "mediapipe/framework/calculator.proto";

message CombinedPredictionCalculatorOptions {
  extend mediapipe.CalculatorOptions {
    optional CombinedPredictionCalculatorOptions ext = 483738635;
  }

  message Class {
    optional string label = 1;
    optional float score_threshold = 2;
  }

  // List of classes with score thresholds.
  repeated Class class = 1;

  // Default score threshold applied to a label.
  optional float default_global_threshold = 2 [default = 0];

  // Name of the background class whose input scores will be ignored while
  // thresholding.
  optional string background_label = 3;
}
